### PR TITLE
[F] Metis: register annotate with iris (#102)

### PR DESCRIPTION
**Pull request by DrWilliamsSteven**
This PR is to:
* remove wrapper function
* move annotate to src folder
* register annotate with iris
* return nothing if no ontology match found

Closes repositive/agora#102

### CODE REVIEW

File: src/service.ts [hunk 1]
```diff
@@ -25,4 +26,10 @@ export default async function init({
       };
     }
   });
+
+  iris.register({
+    pattern: 'action.annotate.metis',
+    handler: annotate
+  });
```
Comment (DrWilliamsSteven) [inline src/service.ts]: @Istar-Eldritch I don't quite understand how this is supposed to be done. This gives me an error.

File: README.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-It takes a dataset and returns an updated dataset with ontology terms for 'assay'/'technology'/'tissue'/'disease' fields.
+It takes a field and term and responds with ontology terms for 'assay'/'technology'/'tissue'/'disease' fields.
 
 1. [http://greekmythology.wikia.com/wiki/Metis](http://greekmythology.wikia.com/wiki/Metis)
 2. [https://en.wikipedia.org/wiki/Metis_(mythology)](https://en.wikipedia.org/wiki/Metis_(mythology))
 
 ### Update Datasets with Ontology Terms
 
-Metis should use Iris to register a `...` pattern ? (TBD).
+Metis uses Iris to register a `action.annotate.metis` pattern.
```
Comment (Istar-Eldritch) [inline README.md]: Let's not use the name of the service in the action. Just the action :slightly_smiling_face:

File: src/annotate.ts [hunk 3]
```diff
@@ -24,10 +13,10 @@ export default async function annotate(field: string, term: string) {
 
   const ontology = ontologyDict[field];
 
-  const excludedTerms = ['n/a', 'na', 'none', 'not available', 'other', 'unavailable', 'unknown', 'unspecified'];
+  const excludedTerms = ['n/a', 'na', 'none', 'not available', 'not specified', 'other', 'unavailable', 'unknown', 'unspecified'];
```
Comment (Istar-Eldritch) [inline src/annotate.ts]: I wouldn't do this filtering here, but is arguable, so yeah. I won't complain if there is a good logic for it.

File: src/service.ts [hunk 4]
```diff
@@ -25,4 +26,10 @@ export default async function init({
       };
     }
   });
+
+  iris.register({
+    pattern: 'action.annotate.metis',
```
Comment (Istar-Eldritch) [inline src/service.ts]: Same as in the previous comment, let's avoid coupling to the name of the service. The action is enough.

File: src/service.ts (hunk 1 above)
Comment (Istar-Eldritch) [reply src/service.ts]: I'll check the PR code locally, to understand what you talk about.

File: src/annotate.ts (hunk 3 above)
Comment (DrWilliamsSteven) [reply src/annotate.ts]: Ideally we won't have those terms at all. I would like to filter them out at the mapper stage, but having a second layer here just helps to catch any that slip through.

**Comment by Istar-Eldritch** [comment]
We need to add to the README the schema for the input of the pattern. I would recommend using JSONSchema and then using it in the handler as to check that the payload is correct. I tried it and if you don't send the correct payload it doesn't notify you of what is going on.

All the rest is correct. Is up  to you to add those modifications to this PR or create a new one.

:+1:

**Comment by DrWilliamsSteven** [comment]
OK thanks @Istar-Eldritch, I'll merge this PR and create an issue to add the schema.

**Pull request merged**